test/testframe.h

```diff
@@ -180,15 +197,16 @@ extern "C" {
  *
  */
 H5TEST_DLL herr_t TestInit(const char *ProgName, void (*TestPrivateUsage)(FILE *stream),
-                           int (*TestPrivateParser)(int argc, char *argv[]), int TestProcessID);
+                           int (*TestPrivateParser)(int argc, char *argv[]), herr_t (*TestSetupFunc)(void),
```

Comment:
Added a run-once setup and cleanup callback for the whole test program. The setup callback is called as part of TestInit() to initialize state as soon as possible and the cleanup callback is called by TestShutdown() to keep the state around as long as possible (for test post-processing or similar).